Gems/ROS2/Code/Source/ROS2SystemComponent.cpp

```diff
@@ -167,6 +170,32 @@ namespace ROS2
         }
     }
 
+    void ROS2SystemComponent::ResetNode()
+    {
+        if (!m_wasNodeReset)
```

Comment:
This is not great as there is no obvious place to call it. Frame Components are many. 

Instead, try this:
1) Determine what exactly we want and need to reset between simulation runs.
2) Have an object with lifetime controlled by Start / Stop simulation events. You might hold it in this System Component but use these events to reset. Make sure lifetime is clear by design and cannot be broken.